src/shared/components/data-table/DataTable.tsx

```diff
@@ -51,7 +51,7 @@ export type TableProps = {
   columns: DataTableColumns;
   state?: Partial<DataTableState>;
   theme: TableTheme;
-  uniqueColumnId?: string; // Values in this column will be used to identify individual rows
+  uniqueColumnId: string; // Values in this column will be used to identify individual rows
```

Comment:
Why is this mandatory? What's unique column id in a BLAST results table?

Reply:
I doubt if it would be any easier to handle row selections & downloads without having a unique column passed from the parent component. 
Another option is to use the position of the current row in the `data` array as the `rowId` but most of the components just use a subset of the data like `currentPageRows`. 
Do you think it would be better to generate the individual row ids within DataTable itself?

Reply:
> I doubt if it would be any easier to handle row selections & downloads without having a unique column passed from the parent component.

One of your earlier arguments was that you want the table component to function both in coordination with the parent, and independently from it.

If that is still your architectural decision, then, for the cases when the DataTable is acting alone, it should be able to decide on the id on its own.

Reply:
Yes, that was my initial thinking. I have now removed `uniqueColumnId` entirely and it doesn't seem to have any major downsides.